Install/install.sh

```diff
@@ -0,0 +1,243 @@
+#!/bin/bash -e
+
+GIT_REPO=https://github.com/TomWhitwell/SlowMovie
+GIT_BRANCH=main
+SKIP_DEPS=false
+
+# color code variables
+RED="\e[0;91m"
+YELLOW="\e[0;93m"
+RESET="\e[0m"
+
+# file paths
+SERVICE_DIR=/etc/systemd/system
+SERVICE_FILE=slowmovie.service
+
+function install_linux_packages(){
+  sudo apt-get update
+
+  sudo apt-get install -y ffmpeg git python3-pip
+}
+
+function install_python_packages(){
+
+
+  sudo pip3 install setuptools -U
+  sudo pip3 install -r $LOCAL_DIR/Install/requirements.txt
+}
+
+function build_python_libraries(){
+  WAVESHARE_DIR=/home/pi/e-Paper
+
+  if [ -d "${WAVESHARE_DIR}" ]; then
+    echo -e "Updating Waveshare Drivers"
+    cd $WAVESHARE_DIR
+    git fetch
+    git pull
+  else
+    echo -e "Installing Waveshare Drivers"
+    git clone https://github.com/waveshare/e-Paper
+  fi
+
+  echo -e "${YELLOW}Be patient — this takes time${RESET}"
+  cd $WAVESHARE_DIR/RaspberryPi_JetsonNano/python/
+  sudo python3 setup.py install
+
+  # return to home directory
+  cd /home/pi
+}
+
+function setup_hardware(){
+  echo "Setting up SPI"
+  if ls /dev/spi* &> /dev/null; then
+      echo -e "SPI already enabled"
+  else
+      if command -v raspi-config > /dev/null && sudo raspi-config nonint get_spi | grep -q "1"; then
+          sudo raspi-config nonint do_spi 0
+          echo -e "SPI is now enabled"
+      else
+          echo -e "${RED}There was an error enabling SPI, enable manually with sudo raspi-config${RESET}"
+      fi
+  fi
+}
+
+function service_installed(){
+  # return 0 if the service is installed, 1 if no
+  if [ -f "$SERVICE_DIR/$SERVICE_FILE" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function install_slowmovie(){
+  FIRST_TIME=1  # if this is a first time install
+
+  if [ "$SKIP_DEPS" = false ]; then
+    # install from apt
+    install_linux_packages
+
+    # configure the hardware
+    setup_hardware
+  else
+    echo -e "Skipping dependency installs, updating SlowMovie code only"
+  fi
+
+  if [ -d "${LOCAL_DIR}" ]; then
+    echo -e "Existing Install Found - Updating Repo"
+    cd $LOCAL_DIR
+    git fetch
+    git checkout $GIT_BRANCH
+    git pull
+
+    # go back to home directory
+    cd /home/pi/
+  else
+    echo -e "No Install Found - Cloning Repo"
+    git clone -b ${GIT_BRANCH} ${GIT_REPO} ${LOCAL_DIR}
+    FIRST_TIME=0
+  fi
+
+  if [ "$SKIP_DEPS" = false ]; then
+    # install any needed python packages
+    install_python_packages
+
+    # install any additional libraries we need to build manually
+    build_python_libraries
+  fi
+
+  cd $LOCAL_DIR
+
+  # check if the service file needs to be updated
+  if (service_installed) && ! (cmp -s "slowmovie.service" "/etc/systemd/system/slowmovie.service"); then
+    sudo cp $SERVICE_FILE $SERVICE_DIR
+    sudo systemctl daemon-reload
+
+    echo -e "Updating SlowMovie service file"
+  fi
+
+  echo -e "SlowMovie install/update complete. To test, run '${YELLOW}python3 slowmovie.py${RESET}'"
```

Comment:
This should have the path to the slowmovie.py file in it, say "from the SlowMovie directory, run..."

When running the installer I ran it from `~` and if I didn't know better, running `python3 slowmovie.py` from there wouldn't work.

Reply:
Added with c84cb168f9094cbce8a3371ac2222f2b8c29afbf